src/trainer/tokenization.py

```diff
@@ -0,0 +1,36 @@
+from typing import List
+from transformers import AutoTokenizer, PreTrainedTokenizer, BatchEncoding
+
+def tokenize(tokenizer: PreTrainedTokenizer, text: str, eos_token: bool=False, max_length: int=2048) -> BatchEncoding:
+    """
+    ## Tokenize Text
+    Tokenize the text with the specified tokenizer.
+
+    Parameters:
+    - tokenizer: PreTrainedTokenizer, The tokenizer to use.
+    - text: str, The text to tokenize.
+
+    Returns:
+    - List[str], The list of tokens.
+    """
+
+    token_list = tokenizer(
+            text, # Prompt.
+            truncation=True, # Truncate the text if it exceeds the maximum length.
+            max_length=max_length, # Maximum length of the tokenized text.
+            padding=False, # No padding.
+        )
+    
+    if eos_token:
+        # Append the EOS token. 
+        # Note that it will append the token even if it exceeds the maximum length.
+        token_list["input_ids"].append(tokenizer.eos_token_id) # type: ignore
```

Comment:
More keys!!! But official package 🫠

Reply:
In this cast, the returned value is a dict which contains token index list (input_ids) and attention mask.
We followed the implementation of official documentation here.